testing/trino-testing/src/main/java/io/trino/testing/TestingConnectorBehavior.java

```diff
@@ -20,6 +20,7 @@
 public enum TestingConnectorBehavior
 {
     SUPPORTS_PREDICATE_PUSHDOWN,
+    SUPPORTS_PREDICATE_MODULO_PUSHDOWN(false),
```

Comment:
are we going to add a separate connector behavior for every function or function/types combination?
I am concerned this is gonna be too fine-grained.

Reply:
btw this should be "modulus"

Reply:
I thought `SUPPORTS_PREDICATE_EXPRESSION_PUSHDOWN` was meant to capture all connector expression based pushdowns. Agreed we shouldn't make it too fine-grained.

Reply:
> all connector expression based pushdowns. 

we can test it coarsely only though
the exact supported functions will vary between connectors.

Reply:
for the ones we currently implement I see no reason why the support should vary between connectors (other than if we chose not to implement it in others).

Another alternative is to have single SUPPORTS toggle but have different test methods which can be overridden as needed.

Reply:
... or we go fine-grained.

surely we want to have some uber-toggle like SUPPORTS_PREDICATE_EXPRESSION_PUSHDOWN
and **IF** we have finer-grain toggles, they should default to the uber one.

Reply:
with connector expressions the fine-grained is unbounded and I don't like that direction. We can start with fine-grained defaulting to higher level one and see what happens over time.

Reply:
Not sure what is the decision on this

Reply:
Sorry about that - some discussion moved offline.
Here's some ideas:

- We can add `SUPPORTS_ARITHMETIC_EXPRESSION_PUSHDOWN` with value defaulting to that of `SUPPORTS_EXPRESSION_PUSHDOWN` (or whatever it's called).
- We add test in BaseJdbcConnectorTest and set `SUPPORTS_ARITHMETIC_EXPRESSION_PUSHDOWN` as `false` in connectors where it's not supported.

In future when we need we can add more fine-grained enums like for MODULO, DIVIDE, MULTIPLY etc. but let's not do that today.

Reply:
I made it:
`SUPPORTS_PREDICATE_ARITHMETIC_EXPRESSION_PUSHDOWN(SUPPORTS_PREDICATE_EXPRESSION_PUSHDOWN),`

In BaseJdbcConnectorTest it's already:
```
            case SUPPORTS_PREDICATE_EXPRESSION_PUSHDOWN:
                // TODO support pushdown of complex expressions in predicates
                return false;
```
So for all connectors it's turned off as we want, however for Postgres:
```
           case SUPPORTS_PREDICATE_EXPRESSION_PUSHDOWN:
                // TODO remove once super has this set to true
                verify(!super.hasBehavior(connectorBehavior));
                return true;
```
And for SqlServer I added:
```
 case SUPPORTS_PREDICATE_ARITHMETIC_EXPRESSION_PUSHDOWN:
return true
```